babel/units.py

```diff
@@ -0,0 +1,241 @@
+
+    .. versionadded:: 2.2.0
+
+    :param value: the value to format.
+    :param measurement_unit: the code of a measurement unit.
+                             Known units can be found in the CLDR Unit Validity XML file:
+                             http://unicode.org/repos/cldr/tags/latest/common/validity/unit.xml
+    :param length: "short", "long" or "narrow"
+    :param format: An optional format, as accepted by `format_decimal`.
+    :param locale: the `Locale` object or locale identifier
+    """
+    locale = Locale.parse(locale)
+
+    q_unit = _find_unit_pattern(measurement_unit, locale=locale)
+    if not q_unit:
+        raise UnknownUnitError(unit=measurement_unit, locale=locale)
+    unit_patterns = locale._data["unit_patterns"][q_unit].get(length, {})
+    formatted_value = format_decimal(value, format, locale)
+    plural_form = locale.plural_form(value)
+    if plural_form in unit_patterns:
+        return unit_patterns[plural_form].format(formatted_value)
+
+    # Fall back to a somewhat bad representation.
+    return '%s %s' % (
+        formatted_value,
+        (get_unit_name(measurement_unit, length=length, locale=locale) or measurement_unit)
+    )
+
+
+def _find_compound_unit(numerator_unit, denominator_unit, locale=LC_NUMERIC):
+    """
+    Find a predefined compound unit pattern.
+
+    Used internally by format_compound_unit.
+
+    >>> _find_compound_unit("kilometer", "hour", locale="en")
+    'speed-kilometer-per-hour'
+
+    >>> _find_compound_unit("mile", "gallon", locale="en")
+    'consumption-mile-per-gallon'
+
+    :param numerator_unit: The numerator unit's identifier
+    :param denominator_unit: The denominator unit's identifier
+    :param locale: the `Locale` object or locale identifier
+    :return: A key to the `unit_patterns` mapping, or None.
+    """
+    locale = Locale.parse(locale)
+    numerator_unit_length = _find_unit_pattern(numerator_unit, locale=locale)
+    denominator_unit_length = _find_unit_pattern(denominator_unit, locale=locale)
+    if not (numerator_unit_length and denominator_unit_length):
+        return None
+    # Since compound units are named "speed-kilometer-per-hour", we'll have to slice off
+    # the quantities from both qualified units.
+    bare_numerator_unit = numerator_unit_length.split("-", 1)[-1]
+    bare_denominator_unit = denominator_unit_length.split("-", 1)[-1]
+    # Now we can try and rebuild a compound unit specifier, then qualify it:
+    return _find_unit_pattern("%s-per-%s" % (bare_numerator_unit, bare_denominator_unit), locale=locale)
+
+
+def format_compound_unit(
+    numerator_value, numerator_unit=None,
+    denominator_value=1, denominator_unit=None,
+    length='long', format=None, locale=LC_NUMERIC
+):
+    """
+    Format a compound number value, i.e. "kilometers per hour" or similar.
+
+    Both unit specifiers are optional to allow for formatting of arbitrary values still according
+    to the locale's general "per" formatting specifier.
```

Comment:
Please refer to http://www.unicode.org/reports/tr35/tr35-general.html#perUnitPatterns here